configs/components/oifs/oifs.yaml

```diff
@@ -1018,20 +1036,81 @@ choose_output:
                                 NAMCT0:
                                         NFRPOS: 12
                                         NFRHIS: 12
+#======================================================================================
+# PREPROCESS: eternal runs
+#======================================================================================
+# TODO: style Jesus not supported yet by these new preprocessing scripts @JOAKIM
 
+# The following choose defines a pseudo_initial_date for OIFS. This allows us to trick
+# OIFS into thinking that, from leg 3, it is always running a 2nd leg, avoiding the
+# memory issues mentioned before. This is done by declaring the pseudo_initial_date
+# as the last start date.
+choose_general.run_number:
+        1:
+                pseudo_initial_date: "${initial_date}"
+                #TODO: prev_pseudo_initial_date: "${initial_date}"
+                start_ndays_source: "${start_ndays}"
+                preprocess_method: "
+                    ${general.esm_function_dir}/components/oifs/change_icm_date.sh
+                        ${thisrun_input_dir}/
+                        ${oifs.input_expid}
+                        ${oifs.input_expid}
+                        ${initial_date!syear!smonth!sday}
+                        ${start_date!syear!smonth!sday}
+                        ${oifs.wam_number}
+                        ${oifs.perturb}
+                        ${oifs.nx}
+                        ${oifs.ensemble_id}"
+        2:
+                pseudo_initial_date: "${initial_date}"
+                #TODO: prev_pseudo_initial_date: "${initial_date}"
```

Comment:
Do we still need this or can this be removed?

Reply:
I'd like to keep it there. That variable might be needed it in case the legs have different lengths.

Reply:
Ah okay, would it be hard to get that in before the merge? I think this is a very common use case. E.g. all of Dima Sidorenkos runs recently have been with changing the length of the run in the middle.

Reply:
Basically to do a restart with a different leg length? I'll try that then.

Reply:
Yes. I think most users will change the length between leg 1 and 2, but it would be nice to also test the cases of changing between leg 2 and 3, as well at between 3 and 4.